packages/editor/src/editor/sidebar/fields/condition-builder/ConditionBuilder.tsx

```diff
@@ -0,0 +1,137 @@
+import { BasicCheckbox, BasicCollapsible, Button, Flex, IvyIcon, Tabs, TabsList, TabsTrigger } from '@axonivy/ui-components';
+import { IvyIcons } from '@axonivy/ui-icons';
+import { useState } from 'react';
+import { ConditionGroup } from './ConditionGroup';
+import type { LogicalOperator } from './Condition';
+
+interface ConditionBuilderProps {
+  onChange: (value: string) => void;
+  apply: () => void;
+}
+
+export const ConditionBuilder = ({ onChange, apply }: ConditionBuilderProps) => {
+  const [isConditionGroupEnabled, setIsConditionGroupEnabled] = useState(false);
+  const [conditionGroups, setConditionGroups] = useState<ConditionGroup[]>([
+    { conditions: [{ argument1: '', operator: '==', argument2: '', logicalOperator: 'and' }], logicalOperator: 'and' }
+  ]);
+
+  const updateCondition = (groupIndex: number, conditionIndex: number, key: string, newValue: string) => {
```

Comment:
stricter type key/value
```suggestion
  const updateCondition = <TKey keyof typeof Condition>(groupIndex: number, conditionIndex: number, key: TKey, newValue: Condition[TKey]) => {
```

Reply:
Thanks👍  However, the previous suggestion contained some errors. I have now implemented it as follows (TKey extends keyof Condition):
```
export const Condition = ({ condition, conditionIndex, groupIndex, conditionsCount, setConditionGroups }: ConditionProps) => {
  const updateCondition = <TKey extends keyof Condition>(
    groupIndex: number,
    conditionIndex: number,
    key: TKey,
    newValue: Condition[TKey]
  ) => {
    setConditionGroups(old => {
      old[groupIndex].conditions[conditionIndex] = { ...old[groupIndex].conditions[conditionIndex], [key]: newValue };
      return old;
    });
  };
```
 is this correct?

Reply:
As I wrote it only here in the comment, the chance is hight that it was wrong. but you get the idea ;)